### ISSUE TITLE
Ambiguous use of 'fetchRequest()'

### ISSUE THREAD

**Issue by alex-taffe** [RoverPlatform/rover-campaigns-ios#105]
On Rover campaigns 3.9.0, we're getting a compile failure. Pinning back to 3.7.7 fixes the issue.
```
/Pods/RoverCampaigns/Sources/Location/Model/Geofence.swift:160:66: [31mambiguous use of 'fetchRequest()'[0m

        let fetchRequest: NSFetchRequest<NSFetchRequestResult> = Geofence.fetchRequest()  as! NSFetchRequest<NSFetchRequestResult>
[36m                    ^[0m
```
Sometimes the error says that `NSFetchRequest<Beacon>` cannot be assigned to type `NSFetchRequest<NSFetchRequestResult>`.

The issue seems to be coming from `Beacon.swift` where `fetchRequest` is returning a `NSFetchRequest<Beacon>` [here](https://github.com/RoverPlatform/rover-campaigns-ios/blob/8f0e629aae7dd6a159e4ab9858c4195e28dda63c/Sources/Location/Model/Beacon.swift#L19) and trying to assign it to a `NSFetchRequest<NSFetchRequestResult>` [here](https://github.com/RoverPlatform/rover-campaigns-ios/blob/8f0e629aae7dd6a159e4ab9858c4195e28dda63c/Sources/Location/Model/Beacon.swift#L123)

**Comment by alex-taffe** [comment]
Appears to be happening in `Geofence.swift` as well

```
Pods/RoverCampaigns/Sources/Location/Model/Beacon.swift:123:73: cannot assign value of type 'NSFetchRequest<Beacon>' to type 'NSFetchRequest<NSFetchRequestResult>'
```

```
Pods/RoverCampaigns/Sources/Location/Model/Geofence.swift:160:75: cannot assign value of type 'NSFetchRequest<Geofence>' to type 'NSFetchRequest<NSFetchRequestResult>'
```

**Comment by orospakr** [comment]
Hey @alex-taffe you guys use Cocoapods, right?

What version of Xcode are you using?

**Comment by alex-taffe** [comment]
Yep, cocoa pods. It's doing it on Xcode 13.1, 12.5.1, and 12.3

**Comment by alex-taffe** [comment]
@orospakr turns out this is actually just happening on all versions of rover campaigns when used on Xcode 13. Is there a timeline for resolving this? It's starting to cause us issues internally

**Comment by orospakr** [comment]
@alex-taffe just to confirm, are you saying this was happening for you with 3.7.7 as well? This isn't a regression in 3.9.0?

**Comment by alex-taffe** [comment]
Correct, seems to be an Xcode 13 thing

**Comment by orospakr** [comment]
@alex-taffe so we've actually been working on it today, and we think we've got a fix in the works.

**Comment by alex-taffe** [comment]
Awesome, thanks!

**Comment by alex-taffe** [comment]
@orospakr any timeline update on this? It's beginning to cause some pretty substantial issues for us

**Comment by orospakr** [comment]
Hey Alex our dev is working on it, and a patch is now in review. Hold
tight; thank you for your patience. 

On January 4, 2022, GitHub ***@***.***> wrote:
> @orospakr <https://github.com/orospakr> any timeline update on this?
> It's beginning to cause some pretty substantial issues for us
>

—
Reply to this email directly, view it on GitHub
<https://github.com/RoverPlatform/rover-campaigns-
ios/issues/105#issuecomment-1005091753>, or unsubscribe
<https://github.com/notifications/unsubscribe-
auth/AAAECSWVJRSDJETNPVU7CTTUUNAERANCNFSM5KBPTP4A>.
You are receiving this because you were mentioned.Message ID:
***@***.***>

**Comment by alex-taffe** [comment]
Thanks

**Comment by orospakr** [comment]
Hey @alex-taffe we have shipped v3.9.1 which should correct the issue.

https://github.com/RoverPlatform/rover-campaigns-ios/releases/tag/v3.9.1

Let me know how you make out :)

**Comment by alex-taffe** [comment]
Thank you!
